File: Difficulty: Easy/Left View of Binary Tree/left-view-of-binary-tree.cpp
/* A binary tree node

struct Node
{
    int data;
    struct Node* left;
    struct Node* right;

    Node(int x){
        data = x;
        left = right = NULL;
    }
};
 */

class Solution {
  public:
    void leftv(Node* root , vector<int> &ans , int level){
        if(root == NULL) return ;
        if(level == ans.size()){
            ans.push_back(root->data);
        }
            leftv(root->left , ans , level+1);
            leftv(root->right , ans , level+1);
        
    }
    vector<int> leftView(Node *root) {
        vector<int> ans;
        if(root == NULL) return ans;
        
        leftv(root , ans , 0);
        
        return ans;
        
    }
};

File: Difficulty: Easy/Left View of Binary Tree/left-view-of-binary-tree.cs
/*
public class Node {
    public int data;
    public Node left;
    public Node right;

    public Node(int key) {
        this.data = key;
        this.left = null;
        this.right = null;
    }
}
*/

class Solution {
    
    void leftView(List<int> ans , Node root , int level){
        if(root == null){
            return;
        }
        if(level == ans.Count){
            ans.Add(root.data);
        }
        leftView(ans , root.left , level+1);
        leftView(ans , root.right , level+1);
        
    }
    public List<int> leftView(Node root) {
        
        List<int> ans = new List<int>();
        if(root == null) return ans;
        
        leftView(ans , root , 0);
        
        return ans;
    }
}